packages/next-intl/src/navigation/react-client/index.tsx

```diff
@@ -2,6 +2,13 @@ export {default as createSharedPathnamesNavigation} from './createSharedPathname
 export {default as createLocalizedPathnamesNavigation} from './createLocalizedPathnamesNavigation';
 export {default as createNavigation} from './createNavigation';
 
+export type {StrictParams} from '../shared/StrictParams';
+export type {
+  HrefOrHrefWithParams,
+  HrefOrUrlObjectWithParams,
+  QueryParams
+} from '../shared/utils';
```

Comment:
Is there a way to get a list of which types need to be exported? Or was this rather trial & error?

Some of these types should have their names reconsidered in case we have to export them. Ideally, we wouldn't have to. In the past, I think I was able to work around exporting some types via `ReturnType<typeof …>` annotations at some places where I saw declaration errors.

Reply:
Unfortunately, it's just trial & error. At least I don't know of a better way to check them. That said, this approach led to some unnecessary exports which I’ve removed in my latest commit.

Reply:
Okay, thanks! I'll also verify the imports again once we can use `example-app-router-playground` to verify that `declaration: true` works.